### ISSUE TITLE
Problem Resubmitting PBS tasks

### ISSUE THREAD

**Issue by drmarkwslater** [ganga-devs/ganga#10]
Hi,
I've got a strange problem with resubmitting PBS tasks.
This is Ganga-6-0-44 here (for the moment I need to stick to the 6.0.x branch).
A simple "task" which demonstrates the problem is ("test_exit.py"):

```python
t = CoreTask()
trf = CoreTransform()
trf.application = Executable()
trf.application.exe = '/bin/sh'
trf.application.args = ['-c', '\"exit 1\"']
trf.backend = PBS()
trf.unit_splitter = GenericSplitter()
trf.unit_splitter.attribute = "application.args"
trf.unit_splitter.values = [ 'unused' ]
trf.abort_loop_on_submit = False
t.appendTransform( trf )
t.float = 100
t.run()
```
which then produces:
```
...
Version: Ganga-6-0-44
...
In [1]:execfile('test_exit.py')

In [2]:Ganga.GPIDev.Lib.Job               : INFO     submitting job 0
Ganga.GPIDev.Lib.Job               : INFO     job 0 status changed to "submitting"
Ganga.GPIDev.Lib.File              : INFO     Preparing Executable application.
Ganga.GPIDev.Lib.File              : INFO     Created shared directory: conf-41d5bb1c-8f15-40cb-8cc4-0f664e3eb3aa
Ganga.GPIDev.Adapters              : INFO     Sending file /bin/sh to shared directory.
Ganga.GPIDev.Lib.File              : INFO     Submitting a prepared application; taking any input files from conf-41d5bb1c-8f15-40cb-8cc4-0f664e3eb3aa
Ganga.GPIDev.Adapters              : INFO     submitting job 0 to PBS backend
Ganga.Lib.Batch                    : INFO     could not match the output and extract the Batch queue name
Ganga.GPIDev.Lib.Job               : INFO     job 0 status changed to "submitted"
Ganga.GPIDev.Lib.Job               : INFO     job 0 status changed to "running"
Ganga.GPIDev.Lib.Job               : INFO     job 0 status changed to "failed"
Ganga.GPIDev.Lib.Job               : INFO     resubmitting job 0
Ganga.GPIDev.Lib.Job               : INFO     job 0 status changed to "submitting"
Ganga.GPIDev.Adapters              : INFO     resubmitting job 0 to PBS backend
Ganga.Lib.Batch                    : WARNING  qsub: illegal -N value
usage: qsub [-a date_time] [-A account_string] [-b secs]
      [-c [ none | { enabled | periodic | shutdown |
      depth=<int> | dir=<path> | interval=<minutes>}... ]
      [-C directive_prefix] [-d path] [-D path]
      [-e path] [-h] [-I] [-j oe] [-k {oe}] [-l resource_list] [-m n|{abe}]
      [-M user_list] [-N jobname] [-o path] [-p priority] [-P proxy_user] [-q queue]
      [-r y|n] [-S path] [-t number_to_submit] [-T type]  [-u user_list] [-w] path
      [-W additional_attributes] [-v variable_list] [-V ] [-x] [-X] [-z] [script]

Ganga.GPIDev.Lib.Job               : ERROR    failed to resubmit job, JobManagerError: error during submit
Ganga.GPIDev.Lib.Job               : WARNING  reverting job 0 to the failed status
Ganga.GPIDev.Lib.Tasks             : ERROR    Couldn't resubmit the job. Deactivating unit.


In [2]:tasks.table()
...
Out[2]:

    # |         Type |                  Name |    State |                       Comment |Jobs: done/  run/  subd/  attd/ fail/ hold/  bad |Float
------------------------------------------------------------------------------------------------------------------------------------------------
    0 |     CoreTask |               NewTask |  running |                               |   1:    0/    0/    0/     --/    1/    0/    0 |  100
  0.0 |CoreTransform |      Simple Transform |  running |                               |   1:    0/    0/    0/     --/    1/    0/    0 |     
------------------------------------------------------------------------------------------------------------------------------------------------

In [3]:exit()
Ganga.Core.MonitoringComponent     : INFO     Stopping the monitoring component...
```

Could you, please, help me.
Thanks in advance,
Best regards,
Jacek.

**Comment by egede** [comment]
Hi Mark,

This is a bit odd. As you can see the submission to PBS works the first time round but not the re-submission. My guess is that the re-submission for some reason is not passing a name to the batch system so you get a command line like "qsub -N -q short job.exe" or something like that. As an argument is expected after the -N it will then fail. It seems like different PBS backends have quite different behaviour, so I would not be surprised if this was not universal.

**Comment by drmarkwslater** [comment]
Fixed with commit af2d409e416d38222a12085c3ba28bcc77eebb0a

**Comment by JacekHoleczek** [comment]
Here's a patch for Ganga 6.0.44:
```
--- python/Ganga/Lib/Batch/Batch.py.old	2015-02-16 15:36:09.000000000 +0100
+++ python/Ganga/Lib/Batch/Batch.py	2015-11-03 19:51:13.342625232 +0100
@@ -250,7 +250,11 @@
             queue_option = queue_option + " " + self.extraopts
                 
         if jobnameopt and job.name != '':
-            queue_option = queue_option + " " + jobnameopt + " " + "'%s'"%(job.name) 
+            # PBS doesn't like names with spaces
+            tmp_name = job.name
+            if self._name == "PBS":
+                tmp_name = tmp_name.replace(" ", "_")
+            queue_option = queue_option + " " + jobnameopt + " " + "'%s'"%(tmp_name) 
                 
         # bugfix #16646 
         if self.config['shared_python_executable']:
```
